src/dbnode/storage/series/buffer.go

```diff
@@ -664,19 +718,19 @@ func (b *BufferBucketVersions) write(
 	value float64,
 	unit xtime.Unit,
 	annotation []byte,
-	wType WriteType,
+	writeType WriteType,
 ) (bool, error) {
-	return b.writableBucketCreate(wType).write(timestamp, value, unit, annotation)
+	return b.writableBucketCreate(writeType).write(timestamp, value, unit, annotation)
 }
 
-func (b *BufferBucketVersions) merge(wType WriteType) (int, error) {
+func (b *BufferBucketVersions) merge(writeType WriteType) (int, error) {
 	res := 0
 	for _, bucket := range b.buckets {
 		// Only makes sense to merge buckets that are writable.
-		if bucket.version == writableBucketVer && wType == bucket.wType {
+		if bucket.version == writableBucketVer && writeType == bucket.writeType {
 			merges, err := bucket.merge()
 			if err != nil {
-				return res, nil
+				return 0, err
```

Comment:
Can you verify this error will get propagated to something that will log it eventually? That would have saved me so much time today

Reply:
Sorry about that. Just verified that this error will always get eventually logged, mostly by passing it all the way up to the fs manager where it is logged.